nixos/modules/services/networking/nomad.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.nomad;
+  dataDir = "/var/lib/nomad";
+  format = pkgs.formats.json { };
+in
+{
+  ##### interface
+  options = {
+    services.nomad = {
+      enable = mkEnableOption "A Distributed, Highly Available, Datacenter-Aware Scheduler";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nomad;
+        defaultText = "pkgs.nomad";
+        description = ''
+          The package used for the Nomad agent and CLI.
+        '';
+      };
+
+      dropPrivileges = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether the nomad agent should be run as a non-root nomad user.
+        '';
+      };
+
+      interface = {
+        advertise = {
+          http = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise http to.";
+          };
+          rpc = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise rpc to.";
+          };
+          serf = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise serf to.";
+          };
+        };
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
+        description = ''
+          Configuration for Nomad. See the <link xlink:href="https://www.nomadproject.io/docs/configuration">documentation</link>
+          for supported values.
+        '';
+      };
+    };
+  };
+
+  ##### implementation
+  config = mkIf cfg.enable {
+    users = {
+      groups.nomad.gid = config.ids.gids.nomad;
+      users.users.nomad = {
+        uid = config.ids.uids.nomad;
+        group = "nomad";
+      };
+    };
+
+    environment = {
+      environment.etc."nomad.json".source = format.generate "nomad.json" cfg.settings;
+      systemPackages = [ cfg.package ];
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${dataDir}' - nomad nomad - -"
+    ];
+
+    systemd.services.nomad = {
```

Comment:
I see a fair bit of drift between your unit and the one [provided by upstream](https://github.com/hashicorp/nomad/blob/9a36ebe2289ceed706bc5dc97472c4e697a9df13/dist/systemd/nomad.service). Any chance you would be interested in utilizing the upstream unit?

I notice upstream only runs as `root`. Is it possible to effectively run this service without `root` privileges, utilizing capabilities instead?

Reply:
Just pushed a change that tries to bring it inline with the upstream unit, thanks for reminding me of it!

Reply:
Are you familiar with the `systemd.packages` option used to directly include upstream unit files?